lib/index.js

```diff
@@ -410,3 +410,16 @@ exports.auto = function(tasks) {
 		});
 	});
 };
+
+exports.retry = function(times, task) {
+	callbackTask = constructIterator(task, 0, true, false);
+	return new Promise(function(resolve, reject) {
+		async.retry.call(async, times, callbackTask, function(err, result) {
```

Comment:
Why .call ?  You're not adjusting 'this' and not using variadic arguments.  Just call async.retry() .